### PR TITLE
Refs #29697 -- Added test for excluding m2m intermediary table.

### PR DESCRIPTION
**Pull request by felixxm**
Fixed in f19a4945e1191e1696f1ad8e6cdc6f939c702728 (@charettes thanks again :star2:).

Ticket [29697](https://code.djangoproject.com/ticket/29697).

### CODE REVIEW

File: tests/queries/tests.py [hunk 1]
```diff
@@ -1173,6 +1173,23 @@ def test_common_mixed_case_foreign_keys(self):
         )
         self.assertTrue(qs.first())
 
+    def test_excluded_intermediary_m2m_table_joined(self):
+        # Three JOINs on "note" column.
```
Comment (charettes) [inline tests/queries/tests.py]: I'm not sure what this comment means.

The first query generates

```sql
SELECT
    "queries_note"."id",
    "queries_note"."note",
    "queries_note"."misc",
    "queries_note"."tag_id"
FROM
    "queries_note"
WHERE
    NOT ("queries_note"."tag_id" IN (
            SELECT
                U2. "tag_id"
            FROM
                "queries_annotation" U2
            WHERE
                U2. "name" = "queries_note"."note")
            AND "queries_note"."tag_id" IS NOT NULL)
    ORDER BY
        "queries_note"."note" ASC
```

And the second

```sql
SELECT
    "queries_note"."id",
    "queries_note"."note",
    "queries_note"."misc",
    "queries_note"."tag_id"
FROM
    "queries_note"
    INNER JOIN "queries_tag" ON ("queries_note"."tag_id" = "queries_tag"."id")
    INNER JOIN "queries_annotation" ON ("queries_tag"."id" = "queries_annotation"."tag_id")
    INNER JOIN "queries_annotation" T4 ON ("queries_tag"."id" = T4. "tag_id")
WHERE ("queries_annotation"."name" = 'a1'
    AND T4. "name" = 'a1'
    AND NOT ("queries_note"."tag_id" IN (
            SELECT
                U2. "tag_id"
            FROM
                "queries_annotation" U2
            WHERE
                U2. "name" = "queries_note"."note")
            AND "queries_note"."tag_id" IS NOT NULL))
ORDER BY
    "queries_note"."note" ASC
```

File: tests/queries/tests.py [hunk 2]
```diff
@@ -1173,6 +1173,23 @@ def test_common_mixed_case_foreign_keys(self):
...
+        self.assertSequenceEqual(
+            Note.objects.filter(~Q(tag__annotation__name=F('note'))),
+            [self.n1, self.n2, self.n3],
+        )
+        self.assertSequenceEqual(
+            Note.objects.filter(
+                tag__annotation__name='a1',
+            ).filter(
+                tag__annotation__name='a1',
```
Comment (charettes) [inline tests/queries/tests.py]: Was this meant to filter twice on `='a1'`?

Comment (felixxm) [reply tests/queries/tests.py]: You're right it is not necessary to reproduce this issue.

File: tests/queries/tests.py (hunk 1 above)
Comment (felixxm) [reply tests/queries/tests.py]: I removed comment.

**Pull request merged**